app/models/media_object.rb

```diff
@@ -569,7 +563,48 @@ def move_bulk documents, params
 
   end
 
+  def _update_permalink_and_dependents
```

Comment:
:+1: I'll give it a shot. I'll also add a non-trivial permalink generator to the test.